homeassistant/components/nest/camera_sdm.py

```diff
@@ -191,7 +191,12 @@ async def async_will_remove_from_hass(self) -> None:
         """Invalidates the RTSP token when unloaded."""
         if self._stream:
             _LOGGER.debug("Invalidating stream")
-            await self._stream.stop_rtsp_stream()
+            try:
+                await self._stream.stop_rtsp_stream()
+            except GoogleNestException as err:
```

Comment:
Is there a more specific exception that could be used?

The source doesn't seem to be directly browsable but here are the options in the tar.gz:
```
class GoogleNestException(Exception):
    """Base class for all client exceptions."""


class SubscriberException(GoogleNestException):
    """Raised during problems subscribing to events and updates."""


class ApiException(GoogleNestException):
    """Raised during problems talking to the API."""


class AuthException(GoogleNestException):
    """Raised due to auth problems talking to API or subscriber."""


class ConfigurationException(GoogleNestException):
    """Raised due to misconfiguration problems."""


class DecodeException(GoogleNestException):
    """Raised when failing to decode a token."""
```

Reply:
The use of `GoogleNestException` in this file should only apply to `ApiException` and `AuthException`.  I think it could make sense to make `AuthException` derive from `ApiException` and catch that here in this file instead, rather than catching both everywhere here. Let me go do some upstream work, then come back to this.

Reply:
Sent https://github.com/home-assistant/core/pull/63224 with some exception handling updates

Reply:
Updated to use a more narrow exception.